snapshot/snapshotfs/upload.go

```diff
@@ -1000,6 +1007,9 @@ func (u *Uploader) Upload(
 		return nil, err
 	}
 
+	cancelScan()
```

Comment:
`cancelScan()` is potentially called twice. Does not seem to be an issue, since IIUC it is idempotent.